src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/CmdLineClient.kt

```diff
@@ -78,27 +87,45 @@ fun zipDirectory(directory: File): ByteArray {
     return byteOutput.toByteArray()
 }
 
-class ResultReceiver() : CompilerFleetMessageReceiver() {
+fun getTaskId(data: ByteArray): String {
+    val messageDigest = MessageDigest.getInstance("SHA-1")
+
+    return String(messageDigest.digest(data))
+}
+
+private val LOGGER = LoggerFactory.getLogger("ResultReceiver")
+
+class ResultReceiver(
+        private val outputFile: File,
+        private val expectedTaskId: String
+) : CompilerFleetMessageReceiver() {
     override fun processMessage(message: PubsubMessage) {
         val result = CompilerFleet.CompilerFleetResult.parseFrom(message.data)
 
-        println(result.taskId)
-        println(String(result.resultBytes.toByteArray()))
+        if (result.taskId != expectedTaskId) {
+            LOGGER.error("Task ids don't match: \nexpected:{}, \nactual:{}", expectedTaskId, result.taskId)
```

Comment:
I don't think it is an error. You just received someone's else result.

By the way, look at the superclass' receiveMessage method. You unconditionally ack any message which means that even if you ignore the result here, other guy will not receive it anyway.

Reply:
Changed to info. Done

Reply:
Do you plan to do anything with ack'ing unexpected results?

Reply:
Maybe, I can republish same message somehow, because other client is waiting for it.
But probably It's not good, when all receivers will republish messages over and over again

Reply:
Why don't you just don't ack it?

Reply:
I don't understand your concern. It already waits forever with that onShutdown future in subscribe method. In my test it won't exit even upon receiving the result.

Reply:
Ok